collect_app/src/androidTest/java/org/odk/collect/android/feature/instancemanagement/InstancesAdbTest.kt

```diff
@@ -0,0 +1,52 @@
+package org.odk.collect.android.feature.instancemanagement
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import org.odk.collect.android.storage.StorageSubdirectory
+import org.odk.collect.android.support.CollectTestRule
+import org.odk.collect.android.support.TestDependencies
+import org.odk.collect.android.support.TestRuleChain
+import org.odk.collect.android.support.pages.MainMenuPage
+import java.io.File
+
+@RunWith(AndroidJUnit4::class)
+class InstancesAdbTest {
+
+    private val testDependencies = TestDependencies()
+    private val rule = CollectTestRule()
+
+    @get:Rule
+    val chain: RuleChain = TestRuleChain.chain(testDependencies).around(rule)
+
+    @Test
+    fun canAddInstanceOnDisk() {
+        rule.startAtMainMenu()
+            .copyForm("one-question.xml")
+            .clickFillBlankForm() // Add form via disk sync
+            .copyInstance("One Question_2021-06-22_15-55-50.xml")
+            .pressBack(MainMenuPage()) // Return to main menu to trigger instance disk sync
+            .clickEditSavedForm(1)
+            .assertText("One Question")
+    }
+
+    @Test
+    fun deletingInstanceOnDisk_andThenOpeningInstance_doesNotCrash() {
+        val mainMenuPage = rule.startAtMainMenu()
+            .copyForm("one-question.xml")
+            .startBlankForm("One Question")
+            .swipeToEndScreen()
+            .clickSaveAndExit()
+
+        val instancesDir = testDependencies.storagePathProvider.getOdkDirPath(StorageSubdirectory.INSTANCES)
+        val instanceDeleted = File(instancesDir).listFiles()[0].deleteRecursively()
+        Assert.assertTrue(instanceDeleted)
+
+        mainMenuPage
+            .clickEditSavedForm(1)
+            .clickOnText("One Question")
```

Comment:
Interestingly, what happens here is that the filled form loads fine, but then has no answers in it. It also appears you can save the form again. We could put more work in to show a specific error message, but I think that should be a follow-up, so we can discuss what to do there.

Reply:
That seems like it could be really confusing. We have reason to believe it would never happen so maybe it doesn't matter. Is it a state we could detect and track analytics for?

Reply:
Yeah we could! We'd need to dig into detecting that in `FormEntryActivity` so it's about as much work as showing an error in the same case.